src/PKSim.BatchTool/Services/SimulationExporter.cs

```diff
@@ -25,44 +29,62 @@ public class SimulationExporter : ISimulationExporter
       private readonly IBatchLogger _logger;
       private readonly IParametersReportCreator _parametersReportCreator;
       private readonly ISimulationResultsExporter _simulationResultsExporter;
+      private readonly ISimulationExportTask _simulationExportTask;
 
-      public SimulationExporter(ISimulationEngineFactory simulationEngineFactory, IBatchLogger logger, IParametersReportCreator parametersReportCreator, ISimulationResultsExporter simulationResultsExporter)
+      public SimulationExporter(ISimulationEngineFactory simulationEngineFactory, IBatchLogger logger, IParametersReportCreator parametersReportCreator,
+         ISimulationResultsExporter simulationResultsExporter, ISimulationExportTask simulationExportTask)
       {
          _simulationEngineFactory = simulationEngineFactory;
          _logger = logger;
          _parametersReportCreator = parametersReportCreator;
          _simulationResultsExporter = simulationResultsExporter;
+         _simulationExportTask = simulationExportTask;
       }
 
-      public void RunAndExport(IndividualSimulation simulation, string outputFolder, string exportFileName, SimulationConfiguration simulationConfiguration, BatchExportMode batchExportMode)
+      public async Task RunAndExport(IndividualSimulation simulation, string outputFolder, string exportFileName, SimulationConfiguration simulationConfiguration, BatchExportMode batchExportMode)
       {
          _logger.AddDebug($"------> Running simulation '{exportFileName}'");
          var engine = _simulationEngineFactory.Create<IndividualSimulation>();
-         engine.RunForBatch(simulation, simulationConfiguration.CheckForNegativeValues);
+         await engine.RunForBatch(simulation, simulationConfiguration.CheckForNegativeValues);
 
-         exportResults(simulation, outputFolder, exportFileName, batchExportMode);
+         await exportResults(simulation, outputFolder, exportFileName, batchExportMode);
       }
 
-      private void exportResults(IndividualSimulation simulation, string outputFolder, string exportFileName, BatchExportMode batchExportMode)
+      private Task exportResults(IndividualSimulation simulation, string outputFolder, string exportFileName, BatchExportMode batchExportMode)
```

Comment:
in theory, writing of those files may happen in // as opposed to sequential. some speed gain here